.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,54 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master", "develop" ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "master", "develop" ]
```

Comment:
I feel like CodeQL could run on any PR, not only `master` or `develop`-backed ones.

Reply:
PR are made more or less exclusively on these branches, but as you prefer.

Also, see the comment: "The branches below must be a subset of the branches above".

Feel free to modify this behavior, I don't have a strong opinion on that.